pkg/chain/ethereum/tbtc_extensions.go

```diff
@@ -0,0 +1,73 @@
+package ethereum
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/keep-network/keep-common/pkg/subscription"
+	"github.com/keep-network/tbtc/bindings/go/contract"
+)
+
+// TBTCEthereumChain represents an Ethereum chain handle with
+// TBTC-specific capabilities.
+type TBTCEthereumChain struct {
```

Comment:
Would it be possible to have this guy living in `pkg/actions`? It looks wrong to me that we are polluting main client's ethereum code with application-specific component/

Reply:
This was my first idea but:
- we need private stuff from `EthereumChain` 
- this fits better with the multi-chain approach we have to support.  If that goes to `pkg/actions` we have to mirror the entire structure of the `pkg/chain` here.

Reply:
Ideally, I think we should have the following structure:
- core Ethereum functions (keep key gen, signing, sortition pool interactions) defined in `pkg/chain/ethereum/`
- core Celo functions defined in `pkg/chain/celo`
- core Algorand functions defined in `pkg/chain/algorand` 
(...)

All chain functions defined by application extensions should be defined elsewhere, for example:
`pkg/extensions(actions)/tbtc/chain/ethereum`
`pkg/extensions(actions)/tbtc/chain/celo`

And so forth.

This is for the multi-chain approach. Regarding private stuff from `EthereumChain`, what stuff do we need? Maybe those pieces should no longer be private.

Reply:
Ok, that structure makes sense. I'll try to align it in that way. Regarding private stuff, we need: `accountKey`, `client`, `nonceManager`, `miningWaiter`, `transactionMutex`. I think these elements should remain private. However, I will try to address that problem somehow.